rcl/src/rcl/wait.c

```diff
@@ -528,12 +528,34 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
   rmw_time_t * timeout_argument = NULL;
   rmw_time_t temporary_timeout_storage;
 
+  // calculate the number of valid (non-NULL and non-canceled) timers
+  size_t number_of_valid_timers = wait_set->size_of_timers;
+  {  // scope to prevent i from colliding below
+    uint64_t i = 0;
+    for (i = 0; i < wait_set->impl->timer_index; ++i) {
+      if (!wait_set->timers[i]) {
+        number_of_valid_timers--;
+        continue;  // Skip NULL timers.
+      }
+      bool is_canceled = false;
+      rcl_ret_t ret = rcl_timer_is_canceled(wait_set->timers[i], &is_canceled);
+      if (ret != RCL_RET_OK) {
+        return ret;  // The rcl error state should already be set.
+      }
+      if (is_canceled) {
+        number_of_valid_timers--;
+        wait_set->timers[i] = NULL;
+      }
+    }
+  }
+
+  bool using_provided_timeout_not_timer_timeout = true;
```

Comment:
With an inverted logic this variable could be named `is_timer_timeout` which sounds a bit more readable / intuitive.

Reply:
yeah we could come up with a good name at the time of the writing. I'll update this to be clearer, thanks for the variable name suggestion